publish-pipeline/content-publish/src/main/scala/org/sunbird/job/content/publish/helpers/ExtractableMimeTypeHelper.scala

```diff
@@ -0,0 +1,279 @@
+  }
+
+  def copyExtractedContentPackage(obj: ObjectData, contentConfig: ContentPublishConfig, extractionType: String, cloudStorageUtil: CloudStorageUtil): Unit = {
+    if (!isExtractedSnapshotExist(obj)) throw new Exception("Error! Snapshot Type Extraction doesn't Exists.")
+    val sourcePrefix = getExtractionPath(obj, contentConfig, "snapshot")
+    val destinationPrefix = getExtractionPath(obj, contentConfig, extractionType)
+    cloudStorageUtil.copyObjectsByPrefix(sourcePrefix, destinationPrefix)
+  }
+
+  private def isExtractedSnapshotExist(obj: ObjectData): Boolean = {
+    extractablePackageExtensions.exists(key => StringUtils.endsWithIgnoreCase(obj.getString("artifactUrl", null), key))
+  }
+
+
+  def getContentBody(identifier: String, readerConfig: ExtDataConfig)(implicit cassandraUtil: CassandraUtil): String = {
+    // fetch content body from cassandra
+    val select = QueryBuilder.select()
+    select.fcall("blobAsText", QueryBuilder.column("body")).as("body")
+    val selectWhere: Select.Where = select.from(readerConfig.keyspace, readerConfig.table).where().and(QueryBuilder.eq("content_id", identifier))
+    logger.info("Cassandra Fetch Query :: " + selectWhere.toString)
+    val row = cassandraUtil.findOne(selectWhere.toString)
+    if (null != row) row.getString("body") else ""
+  }
+
+  def processECMLBody(obj: ObjectData, config: ContentPublishConfig)(implicit ec: ExecutionContext, cloudStorageUtil: CloudStorageUtil): Map[String, AnyRef] = {
+    val basePath = config.bundleLocation + "/" + System.currentTimeMillis + "_tmp" + "/" + obj.identifier
+    val ecmlBody = obj.extData.get.getOrElse("body", "").toString
+    val ecmlType: String = getECMLType(ecmlBody)
+    val ecrfObj: Plugin = getEcrfObject(ecmlType, ecmlBody)
+
+    // localize assets - download assets to local base path (tmp folder) for validation
+    localizeAssets(obj.metadata.getOrElse("identifier","").asInstanceOf[String], ecrfObj, basePath, config)
+
+    // validate assets
+    val processedEcrf: Plugin = new ECMLExtractor(basePath, obj.identifier).process(ecrfObj)
+
+    // getECMLString
+    val processedEcml: String = getEcmlStringFromEcrf(processedEcrf, ecmlType)
+
+    // write ECML String to basePath
+    writeECMLFile(basePath, processedEcml, ecmlType)
+
+    // create zip package
+    val zipFileName: String = basePath + File.separator + System.currentTimeMillis + "_" + Slug.makeSlug(obj.identifier) + ".zip"
+    createZipPackage(basePath, zipFileName)
+
+    // upload zip file to blob and set artifactUrl
+    val result: Array[String] = uploadArtifactToCloud(new File(zipFileName), obj.identifier, None, config)
+
+    // upload local extracted directory to blob
+    extractPackageInCloud(new File(zipFileName), obj, "snapshot", slugFile = true, basePath, config)
+
+    obj.metadata ++ Map("artifactUrl" -> result(1), "cloudStorageKey" -> result(0), "s3Key" -> result(0))
```

Comment:
```suggestion
    obj.metadata ++ Map("artifactUrl" -> result(1), "cloudStorageKey" -> result(0))
```
Stop stamping "s3Key". Let us deprecate and delete this property...